test/Microsoft.AspNet.Mvc.ViewFeatures.Test/ModelStateDictionaryExtensionsTest.cs

```diff
@@ -0,0 +1,336 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using Microsoft.AspNet.Mvc.ModelBinding;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc
+{
+    public class ModelStateDictionaryExtensionsTest
+    {
+        [Fact]
+        public void AddModelError_ForSingleExpression_AddsExpectedMessage()
```

Comment:
Optional but may be worth testing field or variable references in the expression e.g.
``` c#
  var something = "whatever value";
  modelState.AddModelError<TestModel>(model => something, ...);
```
The above would add an error with key "something" and it's a supported case.